cookbooks/aws-parallelcluster-slurm/libraries/emit_chef_error_event.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+#
+# Cookbook:: aws-parallelcluster-slurm
+# Recipe:: emit_chef_error_event
+# Copyright:: 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+module WriteChefError
```

Comment:
What about combine it with  `\aws-parallelcluster-cookbook/libraries/write_chef_error_handler.rb` and move it out of slurm recipes? 
we can check `if node["cluster"]["node_type"]` in `def report` and have different implementation based on headnode and compute nodes.
Alternatively we can enable different handlers based on different node types

Reply:
After discussion, we decide to keep these two handlers separate, but use the same recipe to enable these handlers according to the node type and/or scheduler type. Have updated the code according to this logic.